src/v/cloud_storage/cache_service.cc

```diff
@@ -697,6 +698,13 @@ ss::future<> cache::invalidate(const std::filesystem::path& key) {
 };
 
 ss::future<space_reservation_guard> cache::reserve_space(size_t bytes) {
+    if (_block_puts) {
```

Comment:
I suppose this is all best effort anyway, but should we loop in case `_block_puts` was reset between scheduling points?

Reply:
It's a very unlikely race because of the amount of walltime between disk stats updates, but you are right: this should be a `while` instead of an `if`.  Changed it.